torch/distributed/optim/zero_redundancy_optimizer.py

```diff
@@ -298,29 +259,30 @@ def consolidate_state_dict(self, to: int = 0) -> None:
                     )
                 else:
                     # Receive the optimizer state from the source rank
-                    local_state_dict = _broadcast_object(
-                        empty_messenger,
+                    object_list = [empty_messenger]
+                    distributed_c10d.broadcast_object_list(
+                        object_list,
                         src_rank=global_rank,
                         group=self.group,
                         device=self._default_device,
                     )
                     self._all_state_dicts.append(
-                        _recursive_copy_to_device(local_state_dict, non_blocking=True, device=torch.device("cpu"))
+                        _recursive_copy_to_device(object_list[0], non_blocking=True, device=torch.device("cpu"))
                     )
             else:
                 if rank == self.rank:
                     # Send the optimizer state to the target rank
-                    _ = _broadcast_object(
-                        self.optim.state_dict(),
+                    _ = distributed_c10d.broadcast_object_list(
+                        [self.optim.state_dict()],
                         src_rank=self.global_rank,
                         group=self.group,
                         device=self._default_device,
```

Comment:
The parameter keyword above is `dist_device`, but `device` is used here.

Reply:
Great catch, renamed to dist_device